pkg/client/unversioned/request.go

```diff
@@ -682,7 +775,15 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		}
 		req.Header = r.headers
 
+		time.Sleep(time.Duration(backoff.calculateBackoff(url)) * time.Second)
 		resp, err := client.Do(req)
+		updateURLMetrics(r, resp, err)
+
+		if err != nil {
```

Comment:
I don't have time to look into this too deeply but this is very coupled to this method and that concerns me.  Retry is already too coupled - I don't want to extend that coupling further.  A factoring of this method where the backoff logic is not visible inside this method is much preferred (just that you want to do a backoff using interface X)